### PR TITLE
DOC: Updated GLM docs for freq_weights and fixed some boo-boos.

### PR DESCRIPTION
**Pull request by thequackdaddy**
Ok this is my first college try. I realized that in `families` sometimes the doc is in LaTeX and sometimes, its not. So I did a few more in LaTeX to remind me of my old college days. 

I also noticed a few typos in the docs... like Negative Binomial was clearly incomplete. 

There's probably a few bugs in this still, but I think its better.

### CODE REVIEW

**Comment by josef-pkt** [comment]
One problem with heavy use of Latex is that it becomes difficult to read in plain text, i.e. docstrings in the interpreter or when reading the source.

The general pattern in scipy and related packages, which we largely follow, is to keep Latex as simple as possible or to a minimum in the docstrings and make heavier use of it in tutorials or example, or also in module docstrings.

for example: what does `\mathrm` and `\cdot` dot? Are they necessary?
I was never a heavy raw Latex user, I did almost all my writing with Scientific Word/Workplace.

I need to run the doc build to check, because I don't remember syntax details very well.
One issue: docstrings are interpreted as strings and backslash `\` is an escape character when python handles the strings. To get the backslash to Latex we need either to double them `\\` or to make the docstring into a raw string by adding the `r` prefix, i.e. start the docstring with `r"""`
We don't have strict rules for that, my pattern is to use double backslash if there are just a few, and make it into a raw docstring when there is more Latex.

**Comment by thequackdaddy** [comment]
Ahhh... I hadn't thought of that. I've been using spyder and it renders all the code very nicely regardless. 

That just pretties up the text in my opinion. `\mathrm` makes the text in an equation non-italics, which looks better (in my mind) for things like `llf`, `resid_dev`, and the like. In retrospect, it looks good in the rendered docs, but i could see it being unreadable in command line. I use `\cdot` to a make a multiplicative dot instead of asterisk. 

I'll take another stab where I focus on simplicity more. I had never really understood how Sphinx worked until today, so I got a little carried away.

**Comment by josef-pkt** [comment]
I think in Latex formulas we can use a space for multiplication instead of an explicit dot, $y = X b$ is understandable, and IIRC is what we use. `(X' \Sigma X)^{-1}` which I can still read.
If we write closer to code or code comments, then we need to specify `dot` or nowadays `@` to distinguish numpy elementwise (Hadamard) multiplication from dot multiplication

**Comment by josef-pkt** [comment]
I haven't used that feature much in Spyder because I had it usually set to plain text and show source.

Rendering looks nice in Spyder  (but Latex is not cleaned up properly in some places.
http://www.statsmodels.org/dev/generated/statsmodels.regression.linear_model.RegressionResults.html
and the same in Spyder  with rich text

**Comment by thequackdaddy** [comment]
Point taken. I've simplified the LaTeX a bunch to make the text more readable. Let me know what you think.

File: statsmodels/genmod/families/family.py [hunk 1]
```diff
@@ -345,7 +366,9 @@ def deviance(self, endog, mu, freq_weights=1., scale=1.):
         -----
         If a constant term is included it is defined as
 
-        :math:`deviance = 2*\\sum_{i}(Y*\\log(Y/\\mu))`
+        .. math::
+
+           D = 2 * \sum_i (Y_i * freq\_weights_i * \log(Y_i / \mu_i))/ \phi
```
Comment (josef-pkt) [inline statsmodels/genmod/families/family.py]: style: freq_weights should be in front so we have the standard formula (without freq weights after it all together)
`D = 2 * \sum_i (freq\_weights_i * Y_i * \log(Y_i / \mu_i))/ \phi`

Comment (josef-pkt) [inline statsmodels/genmod/families/family.py]: Note: old version uses `\\`, either that or raw string

File: statsmodels/genmod/families/family.py [hunk 2]
```diff
@@ -505,19 +543,29 @@ def loglike(self, endog, mu, freq_weights=1., scale=1.):
...
 
         If the links is not the identity link then the loglikelihood
         function is defined as
-        llf = sum((`endog`*`mu`-`mu`**2/2)/`scale` - `endog`**2/(2*`scale`) - \
-            (1/2.)*log(2*pi*`scale`))
+
+        .. math::
+
+           llf = \sum_i freq\_weights_i * ((Y_i * \mu_i - \mu_i^2 / 2) / \phi -
+           Y^2 / 2 * \phi - (1/2) * \log(2 * \pi * \phi))
```
Comment (josef-pkt) [inline statsmodels/genmod/families/family.py]: indent either to sum_i or to freq.
Should the be braces `{}` or parenthesis around the entire expression in the sum
I always find it confusing in papers if authors don't do that.

File: statsmodels/genmod/families/family.py [hunk 3]
```diff
@@ -671,13 +730,14 @@ def loglike(self, endog, mu, freq_weights=1., scale=1.):
...
 
         Notes
         --------
-        llf = -1/scale * sum(endog/mu + log(mu) + (scale-1)*log(endog) +\
-              log(scale) + scale*gammaln(1/scale))
-        where gammaln is the log gamma function.
+        .. math::
+
+           llf = -1 / \phi * \sum_i freq\_weights_i * (Y_i / \mu_i+\log(\mu_i)+
+           (\phi -1) * \log(Y) + \log(\phi) + \phi * \ln \Gamma(1 /\phi))
```
Comment (josef-pkt) [inline statsmodels/genmod/families/family.py]: what's `\phi` here
in general I don't like many of the Greek letter (personal taste and not mnenomic)
needs at least a note/definition, e.g. where `$\phi$ is ... and `$\mu` is ...

`mu` for conditional expectation/mean might be the only Greek letter left in the code. (I suggested kicking them out as much as possible)

File: statsmodels/genmod/generalized_linear_model.py [hunk 4]
```diff
@@ -80,8 +84,12 @@ class GLM(base.LikelihoodModel):
         See Parameters.
     family : family class instance
         A pointer to the distribution family of the model.
+    freq_weights : array
+        See Parameters.
     mu : array
         The estimated mean response of the transformed variable.
+    n : array
```
Comment (josef-pkt) [inline statsmodels/genmod/generalized_linear_model.py]: `n` is too short as a name. We need a more explicit name.

(not part of documentation changes)

File: statsmodels/genmod/generalized_linear_model.py [hunk 5]
```diff
@@ -151,16 +160,20 @@ class GLM(base.LikelihoodModel):
...
+        See above.  Note that `endog` is a reference to the data so that if
         data is already an array and it is changed, then `endog` changes
         as well.
     exposure : array-like
         Include ln(exposure) in model with coefficient constrained to 1. Can
         only be used if the link is the logarithm function.
     exog : array
-        See above.  Note that endog is a reference to the data so that if
-        data is already an array and it is changed, then `endog` changes
+        See above.  Note that `exdog` is a reference to the data so that if
```
Comment (josef-pkt) [inline statsmodels/genmod/generalized_linear_model.py]: typo `exdog`

**Comment by josef-pkt** [comment]
Looks fine to me, except for minor style issues. (We don't have strict rules for some of them yet.)

The missing double backslash or `r` for raw strings will still throw off sphinx/latex in some cases, I expect.

Overall, I think the compromise between Latex and code/raw readability is good.

`\_` doesn't looks so nice in raw format, but I guess is necessary to avoid rendering mistakes as in the Regression Results.

**Comment by thequackdaddy** [comment]
Fair enough. 

I'll try to tackle your suggestions tomorrow. Had a busy day at work so wasn't able to contribute as much today. 

It renders fine for my in spyder on Python 3.5

I recall that Python 2 and 3 do Unicode and strings much differently. I guess I'd prefer we do it all identically. Do you prefer raw strings and single backslashes or non-raw and double slashes? Should I test reading it on Python 2.7? I'm not fully up to speed in the differences between the myriad types of strings especially between Python 2 and 3 (raw vs byte vs Unicode or whatever)

**Comment by josef-pkt** [comment]
We don't have unicode in these docstrings, so there is no difference between python 2 and 3 in this case.
Using `r"""` is less noise in the docstrings with longer formulas, so I prefer that to doubling all backslashes. Besides, it's less work.

(When we need unicode, then I think it's handled with specifying the format on top of a module, but I never checked to make sure. It's mainly relevant when we have references with authors that don't have an ASCII name.)

The problem with backslash that is not escaped only raises an error when the letter afterwards is a python string option, based on my experience. But I don't have a list of those memorized, and it's better to be consistent.  (`\t`, `\n`, ....)

**Comment by thequackdaddy** [comment]
Have been a little sidetracked... I'll try to get to this later tonight or tomorrow.

**Comment by josef-pkt** [comment]
@thequackdaddy  thanks, looks good now, merging

**Pull request merged**